File: sdks/csharp/Renamed/Models/JobStatus.cs
using System.Text.Json.Serialization;

namespace Renamed.Sdk.Models;

/// <summary>
/// Status of an async job.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum JobStatus
{
    /// <summary>
    /// Job is waiting to be processed.
    /// </summary>
    [JsonPropertyName("pending")]
    Pending,

    /// <summary>
    /// Job is currently being processed.
    /// </summary>
    [JsonPropertyName("processing")]
    Processing,

    /// <summary>
    /// Job completed successfully.
    /// </summary>
    [JsonPropertyName("completed")]
    Completed,

    /// <summary>
    /// Job failed with an error.
    /// </summary>
    [JsonPropertyName("failed")]
    Failed
}

/// <summary>
/// Response from job status endpoint.
/// </summary>
public sealed class JobStatusResponse
{
    /// <summary>
    /// Unique job identifier.
    /// </summary>
    [JsonPropertyName("jobId")]
    public required string JobId { get; init; }

    /// <summary>
    /// Current job status.
    /// </summary>
    [JsonPropertyName("status")]
    public JobStatus Status { get; init; }

    /// <summary>
    /// Progress percentage (0-100).
    /// </summary>
    [JsonPropertyName("progress")]
    public int? Progress { get; init; }

    /// <summary>
    /// Error message if job failed.
    /// </summary>
    [JsonPropertyName("error")]
    public string? Error { get; init; }

    /// <summary>
    /// Result data when job is completed.
    /// </summary>
    [JsonPropertyName("result")]
    public PdfSplitResult? Result { get; init; }
}

File: sdks/java/src/main/java/to/renamed/sdk/JobStatus.java
package to.renamed.sdk;

/**
 * Status of an asynchronous job.
 */
public enum JobStatus {

    /**
     * Job has been created but not yet started.
     */
    PENDING("pending"),

    /**
     * Job is currently being processed.
     */
    PROCESSING("processing"),

    /**
     * Job completed successfully.
     */
    COMPLETED("completed"),

    /**
     * Job failed with an error.
     */
    FAILED("failed");

    private final String value;

    JobStatus(String value) {
        this.value = value;
    }

    /**
     * Returns the API value for this status.
     *
     * @return the string value used in API responses
     */
    public String getValue() {
        return value;
    }

    /**
     * Parses a string value into a JobStatus.
     *
     * @param value the string value from the API
     * @return the corresponding JobStatus, or null if not recognized
     */
    public static JobStatus fromValue(String value) {
        if (value == null) {
            return null;
        }
        for (JobStatus status : values()) {
            if (status.value.equalsIgnoreCase(value)) {
                return status;
            }
        }
        return null;
    }

    @Override
    public String toString() {
        return value;
    }
}
